### ISSUE TITLE
Performance improvement

### ISSUE THREAD

**Issue by yusukebe** [honojs/node-server#76]
Indeed, the Hono app, when using node-server, is slow. You can see the benchmarks here:

https://github.com/SaltyAom/bun-http-framework-benchmark

However, there are opportunities to improve performance, and we should do them.

**Comment by rafaell-lycan** [comment]
I agree, especially if it performs better than `Express` and `Hapi` or even as well as `Koa` or `Fastify`.

Nothing is perfect, but there's always room to improve 🔥

**Comment by cdtut** [comment]
@yusukebe Is performance the only limitation of hono on node or are there missing functionality or feature compared to bun?

**Comment by kevinvdburgt** [comment]
@yusukebe is the reason known why the node-server seems that 'slow'?

**Comment by yusukebe** [comment]
Hi @kevinvdburgt 

Node.js Incoming/Outgoing Messages and Web Standards Request/Response are incompatible, and the translations cause overhead.

**Comment by marcosrjjunior** [comment]
maybe there is something we can learn from this implementation?

- [node-http-handler](https://github.com/smithy-lang/smithy-typescript/blob/main/packages/node-http-handler/src/node-http-handler.ts#L194)
- [http request](https://github.com/smithy-lang/smithy-typescript/blob/main/packages/protocol-http/src/httpRequest.ts)

not sure if that helps, I'm just sharing
